File: download_test_image.py
import requests
import os
import sys

# Generic test image downloader
# Usage: python download_test_image.py <name> <image_url>
# Example: python download_test_image.py egg "https://images.pexels.com/photos/1556707/pexels-photo-1556707.jpeg?auto=compress&cs=tinysrgb&w=800"

def download_test_image(name, image_url):
    """Download a test image for testing"""
    # Create images directory if it doesn't exist
    os.makedirs("public/images", exist_ok=True)
    
    # Determine file extension from URL or default to jpg
    ext = 'jpg'
    if '.' in image_url:
        url_ext = image_url.split('.')[-1].split('?')[0].lower()
        if url_ext in ['jpg', 'jpeg', 'png', 'webp', 'gif']:
            ext = url_ext
    
    output_path = f"public/images/{name}.{ext}"
    
    # Download the image
    headers = {
        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
    }
    
    try:
        response = requests.get(image_url, headers=headers)
        
        if response.status_code == 200:
            with open(output_path, "wb") as f:
                f.write(response.content)
            print(f"✅ Successfully downloaded {name} image")
            print(f"📸 Image size: {len(response.content):,} bytes")
            print(f"📍 Saved to: {output_path}")
            return True
        else:
            print(f"❌ Failed to download image: {response.status_code}")
            return False
    except Exception as e:
        print(f"❌ Error downloading image: {e}")
        return False

if __name__ == "__main__":
    # Default to egg image if no args provided
    if len(sys.argv) == 1:
        name = "egg"
        url = "https://images.pexels.com/photos/1556707/pexels-photo-1556707.jpeg?auto=compress&cs=tinysrgb&w=800"
    elif len(sys.argv) == 3:
        name = sys.argv[1]
        url = sys.argv[2]
    else:
        print("Usage: python download_test_image.py <name> <image_url>")
        print("Example: python download_test_image.py egg 'https://example.com/egg.jpg'")
        sys.exit(1)
    
    download_test_image(name, url)

File: util/download-test-image.js
const https = require('https');
const http = require('http');
const fs = require('fs');
const path = require('path');

/**
 * Generic test image downloader
 * Usage: node download-test-image.js <name> <image_url>
 * Example: node download-test-image.js egg "https://images.pexels.com/photos/1556707/pexels-photo-1556707.jpeg?auto=compress&cs=tinysrgb&w=800"
 */

function downloadTestImage(name, imageUrl) {
  // Ensure images directory exists
  const imagesDir = path.join(__dirname, 'public', 'images');
  if (!fs.existsSync(imagesDir)) {
    fs.mkdirSync(imagesDir, { recursive: true });
  }

  // Determine file extension
  let ext = 'jpg';
  const urlParts = imageUrl.split('.');
  if (urlParts.length > 1) {
    const possibleExt = urlParts[urlParts.length - 1].split('?')[0].toLowerCase();
    if (['jpg', 'jpeg', 'png', 'webp', 'gif'].includes(possibleExt)) {
      ext = possibleExt;
    }
  }

  const outputPath = path.join(imagesDir, `${name}.${ext}`);
  const file = fs.createWriteStream(outputPath);

  // Choose protocol based on URL
  const protocol = imageUrl.startsWith('https') ? https : http;

  // Download the image
  protocol.get(imageUrl, (response) => {
    response.pipe(file);
    
    file.on('finish', () => {
      file.close();
      console.log(`✅ Successfully downloaded ${name} image`);
      console.log(`📸 Saved to: ${outputPath}`);
      console.log(`📊 Size: ${fs.statSync(outputPath).size.toLocaleString()} bytes`);
    });
  }).on('error', (err) => {
    fs.unlink(outputPath, () => {});
    console.error('❌ Error downloading image:', err.message);
    process.exit(1);
  });
}

// Parse command line arguments
const args = process.argv.slice(2);

if (args.length === 0) {
  // Default to egg image
  const name = 'egg';
  const url = 'https://images.pexels.com/photos/1556707/pexels-photo-1556707.jpeg?auto=compress&cs=tinysrgb&w=800';
  downloadTestImage(name, url);
} else if (args.length === 2) {
  const [name, url] = args;
  downloadTestImage(name, url);
} else {
  console.log('Usage: node download-test-image.js <name> <image_url>');
  console.log('Example: node download-test-image.js egg "https://example.com/egg.jpg"');
  console.log('\nOr run without arguments to download default egg image');
  process.exit(1);
}
